include/pybind11/stl_bind.h

```diff
@@ -326,6 +327,36 @@ template <typename Vector, typename Class_> auto vector_if_insertion_operator(Cl
     );
 }
 
+// Provide the buffer interface for vectors if we have data() and we have a format for it
+// GCC seems to have "void std::vector<bool>::data()" - doing SFINAE on the existence of data() is insufficient, we need to check it doesn't return void
```

Comment:
There's nothing, in theory, to prevent this from working with some std::vector-like custom class (just like `bind_vector`), so I think the check is worthwhile, but it should really just check for what we want instead of checking for *not* being what the current stdlibc++ implementation does.